spec/factories/categories.rb

```diff
@@ -2,4 +2,16 @@
   factory :category do
     sequence(:name) { |n| "category_#{n}" }
   end
+
+  trait :with_books do
+    after(:create) do |book|
+      create_list(:book, 5, :with_author, category: book)
+    end
+  end
+
+  trait :with_book do
```

Comment:
You can pass a transient attribute to the `with_books` trait instead of crating another trait